src/Xamarin.CodeAnalysis/AttributeAnalyzer.cs

```diff
@@ -0,0 +1,222 @@
+				    }
+
+				    if (message == null)
+					    return;
+
+				    if (availability.Message != null)
+					    message = string.Format ("{0}.\n {1}", message, availability.Message);
+
+				    AddIssue (nodeContext, introduced ? introducedRule : deprecatedRule, string.Format ("'{0}' {1}", symbol.Name, message));
+			    },
+			    operationKindsOfInterest
+			);
+		}
+
+		bool TryGetSymbolMergedAttributes (OperationAnalysisContext context, ISymbol symbol, INamedTypeSymbol adviceAttribute, out PlatformAvailability availability)
+		{
+			availability = null;
+
+			MergeSymbolAttributes (context, symbol, adviceAttribute, ref availability);
+
+			if (symbol is IPropertySymbol property) {
+				var syntax = context.Operation.Syntax;
+				var parent = syntax?.Parent;
+				IMethodSymbol method = null;
+
+				switch(parent) {
+				case AssignmentExpressionSyntax assignment:
+					method = (assignment.Right == syntax) ? property.GetMethod : property.SetMethod;
+					break;
+				case PostfixUnaryExpressionSyntax _:
+				case PrefixUnaryExpressionSyntax _:
+					if ((method = property.GetMethod) != null)
+						MergeSymbolAttributes (context, method, adviceAttribute, ref availability);
+
+					method = property.SetMethod;
+					break;
+				case ElementAccessExpressionSyntax _:
+				case ArgumentSyntax _:
+					method = property.GetMethod;
+					break;
+				}
+
+				if (method != null)
+					MergeSymbolAttributes (context, method, adviceAttribute, ref availability);
+			}
+
+			return availability != null && availability.IsSpecified;
+		}
+
+		void MergeSymbolAttributes (OperationAnalysisContext context, ISymbol symbol, INamedTypeSymbol adviceAttribute, ref PlatformAvailability availability)
+		{
+			foreach (var attr in symbol.GetAttributes ()) {
+				if (attr.AttributeClass == adviceAttribute && attr.ConstructorArguments.Length > 0) {
+					AddIssue (context, adviceRule, attr.ConstructorArguments [0].Value?.ToString ());
+				} else {
+					availability = availability.Merge (attr);
+				}
+			}
+		}
+
+		void AddIssue (OperationAnalysisContext context, DiagnosticDescriptor rule, string message) => context.ReportDiagnostic (Diagnostic.Create (rule, context.Operation.Syntax.GetLocation (), message));
+
+		static string GetMinimumOSVersionLine (AdditionalText infoPlistFilePath)
+		{
+			PDictionary dict = PDictionary.FromFile (infoPlistFilePath.Path.ToString ());
+
+			if (dict.TryGetValue ("MinimumOSVersion", out PString value) || dict.TryGetValue ("LSMinimumSystemVersion", out value))
+				return value.Value;
+
+			return String.Empty; // should never reach here; placeholder value just in case
+		}
+
+		static Version GetMinimumDeploymentTargetNumber (AdditionalText infoPlistFilePath)
+		{
+			var minimumOSNumber = GetMinimumOSVersionLine (infoPlistFilePath);
+
+			Version.TryParse (minimumOSNumber, out Version vers);
+			return vers;
+		}
+	}
+}
+
+namespace AttributeAnalyzer.Extensions {
+
+	public static class MyExtensions {
```

Comment:
Looks like this is a useful extension that could go in its own file as `Extensions.cs` and we might add more as we increase scope? If we end up with a few that work on `IOperation`, we can take them out to an `OperationExtensions.cs` instead.